dev-docs/howtoguides/how_to_setup_windows_machine_for_wsl_testing.md

```diff
@@ -0,0 +1,187 @@
+
+After deploying the Bastion, you just need to present your username and password to log into the
+machine.
+
+## Configuring Hyper-V
+
+After logging into the machine through Bastion, you need to configure Hyper-V by following
+this [tutorial](https://learn.microsoft.com/en-us/azure/lab-services/how-to-enable-nested-virtualization-template-vm-using-script?tabs=powershell)
+
+## Installing winget
+
+Each WSL Ubuntu distro has a dedicated installer in the Microsoft store. Therefore, before launching
+any WSL instance, we need to install the installers first. To achieve that, we need to configure
+**winget**.
+
+To install it on the machine, follow this [tutorial](https://winget.pro/winget-install-powershell/)
+
+```
+if you see the following error during installation, restart the machine first:
+
+Add-AppxPackage : Deployment failed with HRESULT: 0x80073D02, The package could not be installed
+because resources it modifies are currently in use. 
+```
+
+## Installing WSL
+
+To install WSL, just run the following command, as an administrator, on powershell:
+
+```console
+$ wsl --install
+```
+
+After that, we need to update WSL to the pre-release version by running:
+
+```console
+$ wsl --update --pre-release
+```
+
+
+```{note}
+We are installing the pre-release version because it allow us to directly run wsl commands
+through SSH. Once that fix is officially release, we will no longer need to install the
+pre-release version.
+```
+
+Now, install a Bionic instance to test the whole WSL set up is working as expected:
+
+```console
+$ winget install --name "Ubuntu 18.04 LTS" --accept-source-agreements --accept-package-agreements --silent
+```
+
+Now, create the instance with:
+
+```console
+$ & ubuntu1804.exe install --root
+```
+
+After that, just test to see if we can run commands on the WSL instance:
+
+```console
+$ wsl -d Ubuntu-18.04 --exec lsb_release -a
+```
+
+## Installing OpenSSH server
+
+To install OpenSSH server, follow these steps:
+
+1) Open *Settings*
+2) Go to *Apps*
+3) Go to *Optional Features*
+4) Go to *Add an optional feature*
+5) Search for *ssh*
+6) Install OpenSSH Server
+
+After installing the Service, open PowerShell and launch it by running:
+
+```console
+$ Start-Service sshd
+```
+
+And set it to automatically start after boot:
+
+```console
+$ Set-Service -Name sshd -StartupType 'Automatic'
+```
+
+### Configure the SSH keys
+
+We need to add a SSH public key into the Windows machine.
+
+To do that, follow these steps:
+
+1) Create a public key to be used in the Windows machine (You can also reuse an existing key if you want to)
+
+2) Set the **UACLIENT_BEHAVE_WSL_IP_ADDRESS** environment variable with the Windows machine IP address. 
+This variable will be used in the integration test, but we will already use it in one of the
+following steps.
+
+2) Send the public key to the Windows machine:
+
+```console
+$ scp PUB_KEY_PATH ubuntu@$UACLIENT_BEHAVE_WSL_IP_ADDRESS:%programdata%/ssh
```

Comment:
This didn't work for me but the following did
```
scp .ssh/wsl_test_rsa.pub ubuntu@$UACLIENT_BEHAVE_WSL_IP_ADDRESS:C:\\ProgramData\\ssh
```

Reply:
both work for me

Reply:
I am changing the option that works for everyone